openwisp_controller/pki/tests/test_models.py

```diff
@@ -5,13 +5,18 @@
 
 from openwisp_users.tests.utils import TestOrganizationMixin
 
-from . import TestPkiMixin
+from ...config.models import Config, Device, Template, Vpn
+from ...config.tests import CreateConfigTemplateMixin, TestVpnX509Mixin
 from ..models import Ca, Cert
 
 
-class TestModels(TestCase, TestPkiMixin, TestOrganizationMixin):
+class TestModels(TestCase, TestVpnX509Mixin, TestOrganizationMixin, CreateConfigTemplateMixin):
```

Comment:
you need to learn the concept of modules and how they interact otherwise you will never understand openwisp.

The PKI app of openwisp-controller (which handles certification authorities and certificates) is not aware of the internal logic of the Config app (which handles device configurations), so adding this test in the PKI app is a violation of this concept.

I think the best place to put this test is here: https://github.com/openwisp/openwisp-controller/blob/master/openwisp_controller/config/tests/test_template.py

(the Config app instead is aware of the PKI app because it needs it as a dependency so that's fine)